torch/testing/_internal/opinfo/core.py

```diff
@@ -2542,17 +2542,40 @@ def __init__(
 
 
 def sample_inputs_foreach(
-    self, device, dtype, N, *, noncontiguous=False, same_size=False, low=None, high=None
+    self,
```

Comment:
not for this PR, but this function is becoming rather cumbersome, with mutually exclusive options (such as `same_size` and `zero_size`). I think it would make sense to make a lower level function that accepts tuple of sizes, and maybe wrappers that call it.